File: read_file.go
// Go program to scan the words
package main

import (
  "fmt"
  "os"
  "bufio"
  "log"
)

func main() {

  file, err := os.Open("search.txt")
  if err != nil {
    log.Fatal(err)
  }

  Scanner := bufio.NewScanner(file)
  Scanner.Split(bufio.ScanWords)

  for Scanner.Scan() {
    fmt.Println(Scanner.Text())
  }
  if err := Scanner.Err(); err != nil {
    log.Fatal(err)
  }
}

File: read_file.pl
use strict;
use warnings;

my @words;
open (my $inFile, '<', 'search.txt') or die $!;

while (<$inFile>) {
  chomp;
  @words = split(' ');
  foreach my $word (@words) {
      printf "%s\n", $word;
  }
}
close ($inFile);